Engine/ResourceManagement/Importer/ModelImporters/AnimationImporter.cpp

```diff
@@ -58,24 +59,21 @@ void AnimationImporter::GetCleanAnimation(const aiNode* root_node, const aiAnima
 	for (auto& channel_set : aiNode_by_channel)
 	{
 		std::map<size_t, float4x4> channel_transform;
-		float animation_duration = animation->mDuration;
 		for (auto channel : channel_set.second)
 		{
 			float4x4 pre_transform;
-			GetAcumulatedAssimpTransformations(channel, channel_set.second,root_node, pre_transform);
-			GetChannelTransform(pre_transform, channel, animation_duration,channel_transform);
+			GetAcumulatedAssimpTransformations(channel, channel_set.second, root_node, pre_transform);
+			GetChannelTransform(pre_transform, channel, animation_duration, channel_transform);
 		}
 
-		for (size_t i = 0; i <= animation_duration; ++i)
+		for (size_t i = 0; i < animation_duration; ++i)
 		{
-			float3 euler_rotation = channel_transform[i].ToEulerXYX();
-			Quat rotation = Quat::FromEulerXYX(euler_rotation.x, euler_rotation.y, euler_rotation.z);
+			float3 euler_rotation = channel_transform[i].ToEulerXYZ();
+			Quat rotation = Quat::FromEulerXYZ(euler_rotation.x, euler_rotation.y, euler_rotation.z);
```

Comment:
I don't understand how this was working xD

Reply:
: D